### PR TITLE
Fix Stalled Upload 

### PR DESCRIPTION
**Pull request by peterjan**
I noticed an upload being stalled after not having enough candidate workers to finish the upload. Stack trace indicates `uploadSlab` can't send a response down the response channel, causing the waitgroup to wait indefinitely and never fail the upload. I think it's easiest to get rid of the waitgroup all together and use a mutex to wait for all responses and go ahead and close the channel.

### CODE REVIEW

File: worker/upload.go [hunk 1]
```diff
@@ -292,10 +293,17 @@ func (mgr *uploadManager) Upload(ctx context.Context, r io.Reader, rs api.Redund
 	defer close(nextSlabChan)
 
 	// create the response channel
-	var wg sync.WaitGroup
+	var ongoingUploads uint64
 	respChan := make(chan slabUploadResponse)
 	defer func() {
-		wg.Wait()
+		// wait for ongoing uploads to send their response down the channel before closing it
+		if atomic.LoadUint64(&ongoingUploads) > 0 {
```
Comment (ChrisSchinnerl) [inline worker/upload.go:300]: I think there is a race here. Theoretically this `if` can trigger right before the counter is decremented. Then we are in a spot where no response will be sent over the channel anymore to trigger the `for` loop.

Comment (peterjan) [reply worker/upload.go]: hm ok I'll revert to my original idea of not necessarily `return`'ing on the first error we encounter but instead keep it around and return when ongoingUploads is 0

**Comment by peterjan** [comment]
converting it to `DRAFT` until I've tested it on my node

**Comment by peterjan** [comment]
@ChrisSchinnerl alternative is not to close the channel, I'd probably prefer that

**Review by ChrisSchinnerl — APPROVED** [review]

**Pull request merged**